### PR TITLE
Show events in sidebar chronologically, and let user show/hide timestamps

### LINKED ISSUE #289: simulator - checkbox to show/hide timestamps?
Would be helpful to free up some horizontal space if there was an option to show/hide timestamps.

image

### LINKED ISSUE #295: simulator - change log ordering?
@emmby thoughts on changing the event log ordering to be normal (not reversed), but also auto scroll to show the latest? It would then function more like a traditional console output.

I've been playing around with the simulator and find it a bit hard to follow the log flow when reading upwards.

The changes are simple:

```diff
diff --git a/test-gen-2/src/LightSm.html b/test-gen-2/src/LightSm.html
index e00e740..48791e3 100644
--- a/test-gen-2/src/LightSm.html
+++ b/test-gen-2/src/LightSm.html
@@ -82,7 +82,7 @@
 
       .console tbody {
         display: flex;
-        flex-direction: column-reverse;
+        flex-direction: column;
         font-family: monospace;
       }
 
@@ -244,6 +244,8 @@ globalThis.light_yellow = ()=>{ addHistoryRow(new Date(), "Called mock light_yel
             row.appendChild(timeCell);
             row.appendChild(eventCell);
             document.querySelector('tbody').appendChild(row);
+            const logDiv = document.querySelector('div.history');
+            logDiv.scrollTop = logDiv.scrollHeight;
         }
 
         var sm = new LightSm();
```

### PR DESCRIPTION
**Pull request by emmby**
- add a settings button that lets you enable/disable the timestamps
- order events chronologically instead of reverse-chronologically, closes #289
- non-mac devices should no longer show scrollbars (overflow: auto instead of scroll)

Closes #295

### CODE REVIEW

File: src/StateSmith/Output/Sim/HtmlRenderer.cs [hunk 1]
```diff
@@ -136,15 +190,24 @@ public static void Render(StringBuilder stringBuilder, string smName, string moc
...
     <div class=""pane sidebar"">
         <div id=""buttons"">
-            <div class=""titlebar"">Events</div>
+            <div class=""titlebar"">Events            
+              <div class='dropdown'>
+                <span id='dropbtn' class='titlebar-icon dropbtn'>settings</span>
+                <div id='myDropdown' class='dropdown-content'>
+                  <div class='dropdown-item'>
+                    <input type='checkbox' id='timestamps' name='timestamps' value='Timestamps'>
+                    <label for=timestamps'>Timestamps</label>
```
Comment (adamfk) [inline src/StateSmith/Output/Sim/HtmlRenderer.cs:199]:
```suggestion
                    <label for='timestamps'>Timestamps</label>
```
The settings dropdown is great! I played around with enabling/disabling the timestamps and noticed clicking the label didn't seem to work. Looks like missing a single quote.

File: src/StateSmith/Output/Sim/HtmlRenderer.cs [hunk 2]
```diff
@@ -112,8 +131,10 @@ public static void Render(StringBuilder stringBuilder, string smName, string moc
       }
 
       .history {
-        margin-top: 30px;
-        overflow: scroll;    
+        margin-top: 30px;       
+        display: flex;
+        overflow: auto;    
+        flex-direction: column-reverse;
```
Comment (adamfk) [inline src/StateSmith/Output/Sim/HtmlRenderer.cs:137]: this is such a cool trick! No js at all for perfect scroll behavior. No need for an autoscroll button. Very cool!

File: src/StateSmith/Output/Sim/HtmlRenderer.cs (hunk 1 above)
Comment (adamfk) [reply src/StateSmith/Output/Sim/HtmlRenderer.cs]: @emmby I tried applying a suggestion to fix typo, but realized that the code gen should be run again anyways. Can you pull, rerun code gen and then push?

Everything else looks great!

Comment (emmby) [reply src/StateSmith/Output/Sim/HtmlRenderer.cs]: done!
